lib/iris/util.py

```diff
@@ -696,22 +696,39 @@ def _build_full_slice_given_keys(keys, ndim):
     return full_slice
 
 
-def _slice_data_with_keys(data, keys, n_parent_dims):
-    # Combine _build_full_slice_given_keys and column_slices_generator to index
-    # any array-like object with arbitrary keys using an orthogonal indexing
-    # interpretation.
-    # This means that both numpy and lazy data will index in the same way.
-    # Return a dimension map, and the indexed data.
-    #
-    # NOTE: by using slicing on only one index at a time, this also generally
-    # avoids copying the data, except when a key contains a list of indices.
-    full_slice = _build_full_slice_given_keys(keys, n_parent_dims)
-
-    dims_mapping, slices_iter = column_slices_generator(full_slice,
-                                                        n_parent_dims)
+def _slice_data_with_keys(data, keys):
+    """
+
+    Index an array-like object with orthogonal keys.
+
+    This enforces an orthogonal interpretation of indexing, which means that
+    both 'real' (numpy) arrays and other array-likes index in the same way,
+    instead of numpy arrays doing 'fancy indexing'.
+
+    Returns (dim_map, data_region), where :
+
+    *  dim_map (dict) :
+        A dimension map, as for :func:`column_slices_generator`.
```

Comment:
> Doesn't the dimension map come from column_slices_generator, rather than being used for it?
> ... It might also be worth describing what the dimension map does, or how/why it might be used?

By "as for" I mean "as it is for" -- so the use is described there, for the other routine.
As this calls the other routine, and it is public, I would much prefer to simply defer to that description.